plugins/rbac-backend/src/service/enforcer-delegate.ts

```diff
@@ -92,18 +77,24 @@ export class EnforcerDelegate {
 
   async addPolicies(
     policies: string[][],
-    source: Source,
     externalTrx?: Knex.Transaction,
   ): Promise<void> {
     const trx = externalTrx || (await this.knex.transaction());
+
+    const policiesToAdd: string[][] = await policies.reduce<
+      Promise<string[][]>
+    >(async (accP: Promise<string[][]>, p: string[]) => {
+      const acc = await accP;
+      const hasPolicy = await this.enforcer.hasPolicy(...p);
```

Comment:
Noticed we are duplicating the `hasPolicy` check a few times. We have it here, in the `processPolicies` right [before](https://github.com/janus-idp/backstage-plugins/blob/0d7ab0691878579c35b3d75a2cbd3bc1a5919f3b/plugins/rbac-backend/src/service/policies-rest-api.ts#L298C44-L298C59) the `addPolicies` call in `policies-rest-api.ts`, and again we are calling `processPolicies` [before](https://github.com/janus-idp/backstage-plugins/blob/0d7ab0691878579c35b3d75a2cbd3bc1a5919f3b/plugins/rbac-backend/src/service/policies-rest-api.ts#L378) the `updatePolicies` call that will end up calling `addPolicies`.

I think we should find a way to simplify it so that we are not performing duplicate checks.